### PR TITLE
[RFC] log: add reference url to rule error log entries

### PR DESCRIPTION
**Pull request by jmtaylor90**
Signed-off-by: jason taylor <jtfas90@gmail.com>

Make sure these boxes are signed before submitting your Pull Request -- thank you.

- [x] I have read the contributing guide lines at https://redmine.openinfosecfoundation.org/projects/suricata/wiki/Contributing
- [x] I have signed the Open Information Security Foundation contribution agreement at https://suricata-ids.org/about/contribution-agreement/
- [x] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
https://redmine.openinfosecfoundation.org/issues/569

Describe changes:
- Use already implemented documentation code to enhance rule error log entries. 
-
-

[PRScript](https://redmine.openinfosecfoundation.org/projects/suricata/wiki/PRscript) output (if applicable):

### CODE REVIEW

**Comment by jmtaylor90** [comment]
Looking for comments on whether or not this will work before going through and updating all the appropriate spots in the code. Thanks in advance!

File: src/detect-content.c [hunk 1]
```diff
@@ -395,8 +395,9 @@ _Bool DetectContentPMATCHValidateCallback(const Signature *s)
         uint32_t right_edge = cd->content_len + cd->offset;
         if (cd->content_len > max_right_edge) {
             SCLogError(SC_ERR_INVALID_SIGNATURE,
-                    "signature can't match as content length %u is bigger than dsize %u.",
-                    cd->content_len, max_right_edge);
+                    "signature can't match as content length %u is bigger than dsize %u.\n"
```
Comment (victorjulien) [inline src/detect-content.c]: No newline chars please!

Comment (jmtaylor90) [reply src/detect-content.c]: okay, will remove in next version

**Comment by victorjulien** [comment]
I like the URL addition. I wonder though if we could automate this somehow? Maybe print the url from detect-parse.c:742, though I don't really like adding another output line.

**Comment by jmtaylor90** [comment]
> I like the URL addition. I wonder though if we could automate this somehow? Maybe print the url from detect-parse.c:742, though I don't really like adding another output line.

I was wondering too, which SCLog level do we want to print this sort of detail in? error? debug?

**Comment by jmtaylor90** [comment]
Reworking

**Pull request closed without merging**